Composite/Composite.Cs.Tests/CataTests.cs

```diff
@@ -100,5 +94,14 @@ public void InvalidParametersTest()
                 goodSource.Cata<int, string>(badRules);
             });
         }
+
+        [Fact]
+        public void TrivialScenarioTest()
+        {
+            var noGoSource = Enumerable.Range(1, 10).AsLimited(0);

```

Comment:
A little bit strange value's name.

Reply:
@v-ilin Strange as it is, it's what it is. This is a special enumerable. If our code asks even for one element from this source an exception will be thrown.